src/kaleidoscope/key_defs.h

```diff
@@ -212,7 +212,7 @@ class Key {
   // used in a conceptually different way: to get a different symbol in the
   // output. We don't normally think "type `shift`+`1`"; we think "type `!`".
   constexpr bool isKeyboardShift() const {
-    return (isKeyboardModifier() &&
+    return (isKeyboardKey() &&
```

Comment:
Unfortunately, though this looks right, it's not.  The trouble is that there are plugins that need to distinguish between a "shift that is part of a modifier key" and a "shift flag that is part of a symbol".  For purposes of this test, `LSHIFT(Key_RightAlt)` and `RALT(Key_LeftShift)` both need to return `true`.  `LSHIFT(Key_1)`, on the other hand, returns `false`.

If there isn't a OneShot test that fails because of this change, that's an oversight on my part; there should be.

Reply:
mm. This wasn't super-obvious from the code.
I wonder if we want a function to encapsulate "             ((flags_ & SHIFT_HELD) != 0)))"